includes/core/classes/class-setup.php

```diff
@@ -78,6 +78,8 @@ protected function setup_hooks() {
 	 * @return void
 	 */
 	public function activate_gatherpress_plugin() {
```

Comment:
```suggestion
	public function activate_gatherpress_plugin() {
	$gp_settings = [];
```
To @mauteri 's point, the variable should be defined(Or I think you'll see a bunch of undefined offset warnings in the logs) and in this case, I think it just needs a default value.  I'm pretty sure  `get_option( 'gp_general' )` won't work at this point because this happens on plugin activation. So I think just setting the default to an empty array is the best solution here.

Reply:
Ok got it... I don't think this should be on activation hook as new options will then require the same sort of treatment and won't be backwards compatible. This should use the default if not saved to the database. I can take a look at this later and make a recommendation.

Reply:
@jmarx @mauteri The variable is set on activation. No errors/warnings/notices are thrown. Don't need to declare en empty array. Just set what you can. The other settings can't be set until the pages exist and will require that the settings page is visited and options updated. The checkbox, however, will be checked and then saved again when the other settings are saved.